src/target_registry.cpp

```diff
@@ -135,3 +140,143 @@ const BitcodeLib *TargetLibRegistry::getISPCTargetLib(ISPCTarget target, TargetO
     }
     return nullptr;
 }
+
+// Print user-friendly message about supported targets
+void TargetLibRegistry::printSupportMatrix() const {
+    // Vector of rows, which are vectors of cells.
+    std::vector<std::vector<std::string>> table;
+
+    // OS names row
+    std::vector<std::string> os_names;
+    os_names.push_back("");
+    for (int j = (int)TargetOS::windows; j < (int)TargetOS::error; j++) {
+        os_names.push_back(OSToString((TargetOS)j));
+    }
+    table.push_back(os_names);
+
+    // Fill in the name, one target per the row.
+    for (int i = (int)ISPCTarget::sse2_i32x4; i < (int)ISPCTarget::error; i++) {
```

Comment:
Do we have some guarantee that we will not change the order of targets in enums and sse2_i32x4 will be always the 1st one?
Maybe it makes sense to have "none" as the 1st value for targets and OSes enums if it does not introduce many changes.

Reply:
Iterating through enum values is ugly. I suggest keeping it this way. I we have more places, where we have to iterate through these enums, we'll need to come up with a better way.